### ISSUE TITLE
Loads wrong page on button click.

### ISSUE THREAD

**Issue by jregen** [cypress-io/cypress#5878]
### Current behavior:

It ends up at this URL at end of test:
https://rahulshettyacademy.com/seleniumPractise/__/#/cart

### Desired behavior:

By the end of the test the browser should be at this URL:
https://rahulshettyacademy.com/seleniumPractise/#/cart
(That is what happens if you run the test manually)

### Steps to reproduce: (app code and test code)
Test code:
```js
/// <reference types="Cypress" />

describe('My Second Test Suite', function () {
    it('should do stuff', function () {
        cy.visit("https://rahulshettyacademy.com/seleniumPractise/#/")
        cy.get('.search-keyword').type('ca')
        cy.wait(2000)
        cy.get('.products').as('productLocator')
        cy.get('@productLocator').find('.product').each(($el, index, $list) => {
            const textVeg = $el.find('h4.product-name').text()
            if (textVeg.includes('Cashews')) {
                $el.find('button').click()
            }
        })
        cy.get('.cart-icon > img').click()
        cy.wait(2000)
        cy.contains('PROCEED TO').click()
        cy.log('got here')
        
    })
})
```

Cypress.json:
```js
{
    "chromeWebSecurity": false
}
```

I don't have the application code, however, manually it works fine.

### Versions

cypress: 3.7.0
mac os:  10.14.6 (Mojave)
Both Chrome 78  Electron73 are failing same way.

NOTE:
This is from a udemy course called "Cypress - Modern Automation Testing from Scratch"
It works for the instructor with Electron 61.

**Comment by jennifer-shehane** [comment]
This appears to be a duplicate of https://github.com/cypress-io/cypress/issues/3975 Going to close this issue and post my findings on your example over there.

**Comment by jennifer-shehane** [comment]
This fix is available starting in [`4.6.0`](https://docs-staging.cypress.io/guides/references/changelog.html#4-6-0) as an [experiment](https://on.cypress.io/experiments) which you can access by setting this config option in your `cypress.json` or elsewhere:

```json
{
	"experimentalSourceRewriting": true
}
```

The fix is experimental, so there may be some situations where the this is not fixed. 

If you're still this issue while setting the `experimentalSourceRewriting` to true in 4.6.0 - [open a new issue](https://github.com/cypress-io/cypress/issues/new) with a reproducible example + screenshots, etc - filling out our issue template.
